examples/bls12-381-hash/src/bls12-381-hash.rs

```diff
@@ -325,18 +327,869 @@ fn g2_clear_cofactor(p: G2) -> G2 {
+        ]
+    ))));
+
+    yden_k[9usize] = Fp::from_byte_seq_be(&ArrFp::to_be_bytes(&ArrFp(secret_array!(
+        U64,
+        [
+            0x167a55cda70a6e1cu64, 0xea820597d94a8490u64, 0x3216f763e13d87bbu64,
+            0x5308592e7ea7d4fbu64, 0xc7385ea3d529b35eu64, 0x346ef48bb8913f55u64
+        ]
+    ))));
+
+    yden_k[10usize] = Fp::from_byte_seq_be(&ArrFp::to_be_bytes(&ArrFp(secret_array!(
+        U64,
+        [
+            0x04d2f259eea405bdu64, 0x48f010a01ad2911du64, 0x9c6dd039bb61a629u64,
+            0x0e591b36e636a5c8u64, 0x71a5c29f4f830604u64, 0x00f8b49cba8f6aa8u64
+        ]
+    ))));
+
+    yden_k[11usize] = Fp::from_byte_seq_be(&ArrFp::to_be_bytes(&ArrFp(secret_array!(
+        U64,
+        [
+            0x0accbb67481d033fu64, 0xf5852c1e48c50c47u64, 0x7f94ff8aefce42d2u64,
+            0x8c0f9a88cea79135u64, 0x16f968986f7ebbeau64, 0x9684b529e2561092u64
+        ]
+    ))));
+
+    yden_k[12usize] = Fp::from_byte_seq_be(&ArrFp::to_be_bytes(&ArrFp(secret_array!(
+        U64,
+        [
+            0x0ad6b9514c767fe3u64, 0xc3613144b45f1496u64, 0x543346d98adf0226u64,
+            0x7d5ceef9a00d9b86u64, 0x93000763e3b90ac1u64, 0x1e99b138573345ccu64
+        ]
+    ))));
+
+    yden_k[13usize] = Fp::from_byte_seq_be(&ArrFp::to_be_bytes(&ArrFp(secret_array!(
+        U64,
+        [
+            0x02660400eb2e4f3bu64, 0x628bdd0d53cd76f2u64, 0xbf565b94e72927c1u64,
+            0xcb748df27942480eu64, 0x420517bd8714cc80u64, 0xd1fadc1326ed06f7u64
+        ]
+    ))));
+
+    yden_k[14usize] = Fp::from_byte_seq_be(&ArrFp::to_be_bytes(&ArrFp(secret_array!(
+        U64,
+        [
+            0x0e0fa1d816ddc03eu64, 0x6b24255e0d7819c1u64, 0x71c40f65e273b853u64,
+            0x324efcd6356caa20u64, 0x5ca2f570f1349780u64, 0x4415473a1d634b8fu64
+        ]
+    ))));
+
+    // Computation
+
+    // x_num = k_(1,11) * x'^11 + k_(1,10) * x'^10 + k_(1,9) * x'^9 + ... + k_(1,0)
+    let mut xnum = Fp::ZERO();
+    let mut xx = Fp::ONE();
+    for i in 0..xnum_k.len() {
+        xnum = xnum + xx * xnum_k[i];
+        xx = xx * x;
+    }
+
+    // x_den = x'^10 + k_(2,9) * x'^9 + k_(2,8) * x'^8 + ... + k_(2,0)
+    let mut xden = Fp::ZERO();
+    let mut xx = Fp::ONE();
+    for i in 0..xden_k.len() {
+        xden = xden + xx * xden_k[i];
+        xx = xx * x;
+    }
+    xden = xden + xx;
+
+    // y_num = k_(3,15) * x'^15 + k_(3,14) * x'^14 + k_(3,13) * x'^13 + ... + k_(3,0)
+    let mut ynum = Fp::ZERO();
+    let mut xx = Fp::ONE();
+    for i in 0..ynum_k.len() {
+        ynum = ynum + xx * ynum_k[i];
+        xx = xx * x;
+    }
+
+    // y_den = x'^15 + k_(4,14) * x'^14 + k_(4,13) * x'^13 + ... + k_(4,0)
+    let mut yden = Fp::ZERO();
+    let mut xx = Fp::ONE();
+    for i in 0..yden_k.len() {
+        yden = yden + xx * yden_k[i];
+        xx = xx * x;
+    }
+    yden = yden + xx;
+
+    // x = x_num / x_den
+    let xr = xnum * xden.inv();
+
+    // y = y' * y_num / y_den
```

Comment:
I appreciate that you added a lot of comments!
But here for example I can't find y' anywhere else.